### ISSUE TITLE
Possible (partial) property name duplication in generated cfg

### ISSUE THREAD

**Issue by d-geula** [dmcooller/S2ZonaConfigurator#1]
Hey, I didn't want to clutter up the mod page with this sort of thing but I may have found a weird bug unless I'm being an idiot (very possible). Basically, while I was testing out my automated mod conversion code I noticed that the property `NonAutomaticWeaponShotDelay` was getting generated as `NonAutomaticWeapoNonAutomaticWeaponShotDelay`. Other than the stripped comments and whitespace it's the only difference between my generated Better Gunfights mod and the original ([diff](https://www.diffchecker.com/xDaUC4ni/)).

Here is my JSON mod I used to generate the cfg from ([link](https://pastebin.com/rABm5uuf)) if you need it for testing. I can't find any instances of `NonAutomaticWeapoNonAutomaticWeaponShotDelay` in that file so I don't think it's my automation process that's the issue (who knows though). It's also strange because it's only that property and it's not even duplicating the whole thing.

Hope you don't mind me submitting this issue here. Also, what do you think of my generated JSON mod? I can't tell if needs some improvement with that many changes. I do need to finish implementing the `AddStruct` feature because there are a couple even more complex mods that I can't currently properly generate. Pretty sure I have everything else working other than that and `Replace` which I don't think I'll touch - though it does definitely need to be cleaned up!

**Comment by dmcooller** [comment]
Hi! It's ok to submit issues here. If there are no issues, no one is using the app :)

Yeah, there was a problem with a method that modified the values of multiple properties. Many tabulations were at the end of some `NonAutomaticWeaponShotDelay` properties, breaking the indentation calculation algorithm. I fixed that and changed the logic so that if a similar case occurs, it won't duplicate a part of a property name but insert a wrong indentation, which won't break the config file. So, thanks for pointing it out.

I think trimmed comments are ok, but they can also be handled if needed.

The JSON mod looks good. I don't think there is anything that should be changed. That's a lot of changes though, so your script will greatly help! Also, because of that mod, I've noticed that my app processes actions too slowly, so I optimized some stuff.

So, yeah, check the latest release; and thanks for your involvement.

**Comment by dmcooller** [comment]
Also, it seems I should add an ability to specify the file path at the top level of JSON. Something like that:
```json
{
    "version": "1.0",
    "description": "Auto-generated mod",
    "author": "AutoGenerator",
    "file": "Stalker2/Content/GameLite/GameData/WeaponData/WeaponAttributesPrototypes/NPCWeaponAttributesPrototypes.cfg",
    "actions": [
        {
            "type": "RemoveLine",
            "path": "GunPKP_Korshunov_MG_NPC::AIParameters::BehaviorTypes::Newbie::Long::MinSecondsDelay"
        }
     ]
}
```

That would reduce the line count by 531 in this mod. But I need to think through the syntax more precisely.

**Comment by d-geula** [comment]
Just wanted to let you know that it works fine now and sooo much faster (nice one) to process all those changes.

Also yeah, having the file path at the top would definitely make it easier when manually converting/creating a mod. I don't think it makes much difference if it's automated other than generating less lines but either way it could be a nice addition as long as it's not a giant pain in the ass to implement.

btw should I close this or are you going to? This is first time I've ever actually created an issue on someone else's repo lol.

**Comment by dmcooller** [comment]
I'll close this one. I'm pretty sure you can close it too if you think that the issue is solved

**Comment by dmcooller** [comment]
Solved in version 0.0.4
